packages-native/actions/src/camera/TakePictureAdvanced.ts

```diff
@@ -0,0 +1,225 @@
+ * @param {"NativeMobileResources.PictureSource.camera"|"NativeMobileResources.PictureSource.imageLibrary"|"NativeMobileResources.PictureSource.either"} pictureSource - Select a picture from the library or the camera. The default is to let the user decide.
+ * @param {"NativeMobileResources.PictureQuality.original"|"NativeMobileResources.PictureQuality.low"|"NativeMobileResources.PictureQuality.medium"|"NativeMobileResources.PictureQuality.high"|"NativeMobileResources.PictureQuality.custom"} pictureQuality - Set to empty to use default value 'medium'.
+ * @param {Big} maximumWidth - The picture will be scaled to this maximum pixel width, while maintaing the aspect ratio.
+ * @param {Big} maximumHeight - The picture will be scaled to this maximum pixel height, while maintaing the aspect ratio.
+ * @param {MxObject} pictureData - Additional info about the picture will be stored in this object. Create it before calling this action.
+ * @returns {Promise.<mendix.lib.MxObject>}
+ */
+export async function TakePictureAdvanced(
+    picture?: mendix.lib.MxObject,
+    pictureSource?: PictureSource,
+    pictureQuality?: PictureQuality,
+    maximumWidth?: BigJs.Big,
+    maximumHeight?: BigJs.Big
+): Promise<boolean> {
+    // BEGIN USER CODE
+    // Documentation https://github.com/react-native-community/react-native-image-picker/blob/master/docs/Reference.md
+
+    return new Promise((resolve, reject) => {
+        if (!picture) {
+            return reject(new Error("Input parameter 'Picture' is required"));
+        }
+        if (!picture.inheritsFrom("System.FileDocument")) {
+            const entity = picture.getEntity();
+            return reject(new Error(`Entity ${entity} does not inherit from 'System.FileDocument'`));
+        }
+        if (pictureQuality === "custom" && !maximumHeight && !maximumWidth) {
+            return reject(new Error("Picture quality is set to 'Custom', but no maximum width or height was provided"));
+        }
+        createMxObject("NativeMobileResources.TakePictureData").then((resultObject: any) => {
+            takePicture()
+                .then((response: any) => {
+                    if (!response || !response.uri) {
+                        return resolve(resultObject);
+                    }
+                    storeFile(picture, response.uri).then(pictureTaken => {
+                        resultObject.set("PictureTaken", pictureTaken);
+                        resultObject.set("URI", response.uri);
+                        resultObject.set("IsVertical", response.isVertical);
+                        resultObject.set("Width", response.width);
+                        resultObject.set("Height", response.height);
+                        resultObject.set("FileName", response.fileName);
+                        resultObject.set("FileSize", response.fileSize);
+                        resultObject.set("FileType", response.type);
+                        resolve(resultObject);
```

Comment:
Why line 48 you return the promise and here you resolve the promise?

Reply:
I have done my best to keep the code similar to the original TakePicture. However, TakePicture returns a boolean and the action returns a promise in different points in the code. By wrapping the main bit of the action in another promise I gain control over the moment when to return to caller.

Reply:
And I can then return the result object i.s.o. a boolean.

Reply:
What about await instead then?

Reply:
Then the surrounding function definition must be async to allow await. I tried to mix and match that before, lead to funny results on Android.  Stuff misfires or is out of sequence.

Reply:
But in this case makes sense right.. is a dependency that you are waiting for. Even though we need to test your code before

Reply:
Could a part of your comment be missing after the 'before'?
Are you willing to accept the code as it is now? If it passes the test of course. I know it works well on Android and iOS. I got weird delays especially on Android with other versions, this one looks nice and gets the job done.

Reply:
We will test your original code

Reply:
Can you please provide an estimate when this will be merged and released?